bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/StringValue.java

```diff
@@ -28,20 +29,75 @@
  *
  * @since 1.0.5
  */
-public interface StringValue extends BString, SimpleValue {
+public abstract class StringValue implements BString, SimpleValue {
+
+    final String value;
+    final int nonBmpFlag;
```

Comment:
If we are using a flag, it is the best practice to use an `enum`. But since we have only two types we can use a `boolean` flag